static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +530,328 @@ export function getProteinDescription(data: GraphNodes): string {
+ * (since they're automatically converted to strings anyway).
+ *
+ * References:
+ *  - https://linuxtut.com/en/87ab3f313a75b547c278/
+ *  - https://stackoverflow.com/questions/13315131/enforcing-the-type-of-the-indexed-members-of-a-typescript-object
+ *
+ */
+export function objectFromArray<T, V>(
+  arr: T[],
+  keyFunc: (arrElt: T) => string,
+  valFunc: (arrElt: T) => V
+): { [key: string]: V } {
+  return arr.reduce(
+    (obj, elt) => Object.assign(obj, { [keyFunc(elt)]: valFunc(elt) }),
+    {}
+  );
+}
+
+/**
+ * Given an (unpacked) array of arrays, return its transpose (after trimming arrays to length of shortest one).
+ * This is essentially a TS implementation of python's built-in zip function.
+ *
+ * Adapted from implementation by Matt Mancini
+ *
+ * Thanks also to Carlos Israel Ortiz Garcia and Jack Davison for the below references:
+ *  - https://ncjamieson.com/typing-zipwith/#typing-zip
+ *  - https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types
+ */
+// Recursive elementwise type inference on array of arrays
+type Zip<T> = T extends [Array<infer U>, ...infer Z] ? [U, ...Zip<Z>] : [];
+export function zip<T extends unknown[][]>(...arrays: T): Array<Zip<T>> {
+  const minLength = Math.min(...arrays.map((arr) => arr.length));
+  return Array.from(Array(minLength).keys()).map((_, i) => {
+    return arrays.map((array) => array[i]) as Zip<T>;
+  });
+}
+
+/**
+ * Convert DCID of form bio/<id> to id.
+ * Note: probably should be doing dcid: `bio/${string}`, but currently causes many type errors
+ */
+export function idFromDCID(dcid: string): string {
+  return dcid.replace("bio/", "");
+}
+
+/**
+ * Given id, convert to DCID of form bio/<id>
+ */
+export function dcidFromID(id: string): string {
+  return `bio/${id}`;
+}
+
+/**
+ * Given quantity DCID of the form "<quantityName><quantityValue>", extract and return quantityValue.
+ */
+export function quantityFromDCID(
+  quantityDCID: string,
+  quantityName: string
+): number {
+  if (!quantityDCID.includes(quantityName)) {
+    throw `DCID "${quantityDCID}" does not contain quantity name "${quantityName}"`;
+  }
+  return Number(_.last(quantityDCID.split(quantityName)));
+}
+
+/**
+ * Given interaction DCID of the form bio/{protein1}_{protein2}, return [protein1, protein2]
+ */
+export function proteinsFromInteractionDCID(interactionDCID: string): string[] {
+  const id = idFromDCID(interactionDCID);
+  // danger: assumes neither {protein id}, {species id} contain an underscore
+  const split = id.split("_");
+  if (split.length !== 4) {
+    throw `Unsupported DCID ${interactionDCID}`;
+  }
+  return [`${split[0]}_${split[1]}`, `${split[2]}_${split[3]}`];
+}
+
+/**
+ * Given interactionDCID of the form {protein 1}_{protein 2}, return ID of the same form
+ * but with protein 1, protein 2 in alpha order.
+ */
+function sortInteractionDCID(interactionDCID: string): string {
```

Comment:
Since this is only used for the following function - this could return a `proteins` to avoid the string munging.

Reply:
Oh do you mean something like `[P53_HUMAN, 1433G_HUMAN]`?  If so I think we can delete this function and add a `sorted` flag to `proteinsFromInteractionDCID`.

Reply:
I ended up deleting this helper without adding the `sorted` flag because all use cases so far would work with `sorted=False`.